app/app.go

```diff
@@ -129,13 +131,43 @@ func (app *App) TuneNote(noteID string) error {
 	currentState, err := aNote.Initialise()
 	if err != nil {
 		return fmt.Errorf("Failed to examine system for the current status of note %s - %v", noteID, err)
-	} else if err = app.State.Store(noteID, currentState, false); err != nil {
+	}
+	if reflect.TypeOf(currentState).String() == "note.INISettings" {
```

Comment:
I wonder if it would be possible to avoid some of the complexity here by using interfaces: Maybe `INISettings` could implement an interface which provides methods for extracting the SysctlParams and their values?

Reply:
Methods to extract or set SysctlParameters and their values already exist. So that should not the problem.
My problem here is very special for only the 'vm.dirty_\*' parameters. If you set one of the 'vm.dirty_\*' parameters the counterpart 'vm.dirty_\*' parameter will be set to '0'. To revert such a parameter to its value set before we applied our tuning values means - we need the 'old' value of the counterpart 'vm.dirty_\*' parameter as well, because you can't set a 'vm.dirty_\*' parameter to '0' you only can set the counterpart parameter to a value > '0'
In our Note definition files only the 'vm.dirty_\*' parameters are described which are mentioned in the related SAP Note, but not the counterpart parameters. So when I set up the internal sysctl structure by reading the parameter-value pairs from the Note definition file, I only have one of the needed parameters available. But if I add the counterpart parameter as well, then the options 'verify' or 'simulate' will show up these additional parameters, which are not available in the Note definition files. And this will confuse our customers.
So the changes here fit all my needs. The old counterpart values are saved before I apply the new values, the counterpart parameters will not shown up during 'verify' or 'simulate', but after a revert all 'vm.dirty_\*' parameters will be back to their values before. And it's simpler than to filter the counterpart parameters from the  'verify' or 'simulate' output.

Reply:
I see :) Thank you for all of these very clear explanations! This is a lot more complicated than I thought it would be.